proposals/pr-contrib-guide.md

```diff
@@ -0,0 +1,452 @@
+# Improvements to the AWS CLI Contribution Process
+
+Proposal         | Metadata
+---------------- | -------------
+**Author**       | Kenneth Daily
+**Status**       | Proposed
+**Created**      | 19-October-2021
+
+## Abstract
+
+This document proposes standards and processes to improve and publish community
+expectations for contributions. It defines the process for proposing,
+submitting, and reviewing a contribution. This proposal will make it easier for
+the community to successfully contribute to the AWS CLI.
+
+## Background and Motivation
+
+### Current contribution process
+
+The AWS CLI GitHub repository is available for any user to report bugs, make
+feature requests, and ask guidance questions through GitHub issues. There are
+currently roughly three stages to manage issues in use that are common to open
+source projects: intake, implementation, and review.
+
+#### Intake
+
+GitHub issues are triaged regularly to determine that they are correctly
+categorized and express a real and relevant problem or request. An answer is
+provided as soon as possible to acknowledge or resolve the issue. Feature
+requests are reviewed for general suitability and uniqueness, but are not
+actively commented on by maintainers to move the feature forward to a
+contribution from the community. Users can vote for features via "reactions" on
+the issue. The popularity of an issue is occasionally used to determine which
+features the maintainers will work on.
+
+#### Implementation
+
+Users can open pull requests to propose changes to the code base. There is no
+requirement to work on an existing issue in the GitHub repository or to discuss
+a contribution prior to opening a pull request. Occassionally guidance is
+provided to the contributor to improve the proposed change.
+#### Review
+
+Pull requests are selected for review opportunistically when the maintainers
+have decided separately that a change or feature should be incorporated.
+Reactions and comments can influence the liklihood of prioritizing a
+contribution, but they do not guarantee it will be reviewed. When a feature is
```

Comment:
I think it may be worse than that though, as we don't maintain an exhaustive list of GH issues in the internal queue; there is currently an *ad hoc* mechanism to get something into the queue (adding an unprioritized task) before any prioritization is completed.